### PR TITLE
Introduce DateOfBirth model to encapsulate age and DOB handling

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://www.pivotaltracker.com/story/show/161671487

### CODE REVIEW

**Comment by rakshakhegde** [comment]
Everything looks good. Please just check the one comment left. After that we can merge this.

File: app/src/test/java/org/simple/clinic/patient/DateOfBirthTest.kt [hunk 1]
```diff
@@ -39,4 +39,33 @@ class DateOfBirthTest {
...
+    // given
+    val age = Age(value = 30, updatedAt = Instant.parse("2018-01-01T00:00:00Z"))
+
+    fun estimateAgeAtDate(date: LocalDate): Int {
+      val clock = TestUserClock(date)
+      return DateOfBirth.fromAge(age, clock).estimateAge(clock)
+    }
+
+    // when
+    val `estimated age on the same day` = estimateAgeAtDate(LocalDate.parse("2018-01-01"))
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/patient/DateOfBirthTest.kt]: Good point! I'll do this.

[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/test/java/org/simple/clinic/patient/DateOfBirthTest.kt]: Done.

File: app/src/main/java/org/simple/clinic/patient/DateOfBirth.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      val ageRecordedAtDate = age.updatedAt.atZone(userClock.zone)
+
+      val guessedDateOfBirth = ageRecordedAtDate.minusYears(age.value.toLong()).toLocalDate()
+
+      return DateOfBirth(guessedDateOfBirth, GUESSED)
+    }
+  }
+
+  enum class Type {
+    RECORDED, GUESSED
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/patient/DateOfBirth.kt]: Discussed offline: We will rename the values to `EXACT` and `FROM_AGE` for better clarity.

[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/patient/DateOfBirth.kt]: Done.

**Pull request merged**